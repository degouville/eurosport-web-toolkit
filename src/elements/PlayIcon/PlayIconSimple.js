import React from 'react';
import PropTypes from 'prop-types';
import play from '../../assets/play.svg';

const PlayIconSimple = ({ alt, height, ...props }) => (
  <div {...props}>
    <img src={play} alt={alt} height={height} />
  </div>
);

PlayIconSimple.defaultProps = {
  alt: 'play',
};

PlayIconSimple.propTypes = {
  alt: PropTypes.string,
  height: PropTypes.number.isRequired,
};

export default PlayIconSimple;
