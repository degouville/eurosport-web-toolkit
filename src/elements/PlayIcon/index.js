import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import { rgba } from 'polished';
import { white } from '../../colors';

const icon =
  'data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgNDIgNDIiIHZlcnNpb249IjEuMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48cGF0aCBkPSJNNDEuMzE0MTMxMiwyMi4yNzIxNDE5IEwxMy4yOTQ2MDAxLDM5Ljc3MjE1MTcgQzEyLjU5MTk1NzksNDAuMjEwOTk3MiAxMS42NjY1OTkyLDM5Ljk5NzE0NzggMTEuMjI3NzUzNywzOS4yOTQ1MDU1IEMxMS4wNzg5MTQ1LDM5LjA1NjE5NjkgMTEsMzguNzgwODc1MyAxMSwzOC40OTk5MDU0IEwxMSwzLjQ5OTg4NTggQzExLDIuNjcxNDU4NjcgMTEuNjcxNTcyOSwxLjk5OTg4NTggMTIuNSwxLjk5OTg4NTggQzEyLjc4MDk2OTksMS45OTk4ODU4IDEzLjA1NjI5MTUsMi4wNzg4MDAzMyAxMy4yOTQ2MDAxLDIuMjI3NjM5NSBMNDEuMzE0MTMxMiwxOS43Mjc2NDkzIEM0Mi4wMTY3NzM0LDIwLjE2NjQ5NDggNDIuMjMwNjIyOSwyMS4wOTE4NTM1IDQxLjc5MTc3NzQsMjEuNzk0NDk1NyBDNDEuNjcwOTQ4OCwyMS45ODc5NTYyIDQxLjUwNzU5MTcsMjIuMTUxMzEzMyA0MS4zMTQxMzEyLDIyLjI3MjE0MTkgWiIgZmlsbD0iI2ZmZiI+PC9wYXRoPjwvc3ZnPg==';

const StyledPlayIcon = styled.img`
  box-sizing: border-box;
  height: ${({ iconHeight }) => `${iconHeight}px`};
  padding: ${({ iconHeight }) => `${(iconHeight * 0.45) / 2}px`};
  border: 2px solid ${rgba(white, 0.5)};
  border-radius: 50%;
  background: rgba(6, 8, 37, 0.2);
`;

const PlayIcon = ({ alt, className, height }) => (
  <StyledPlayIcon alt={alt} className={className} src={icon} iconHeight={height} />
);

PlayIcon.defaultProps = {
  alt: 'play',
  className: '',
};

PlayIcon.propTypes = {
  alt: PropTypes.string,
  height: PropTypes.number.isRequired,
  className: PropTypes.string,
};

export default PlayIcon;
