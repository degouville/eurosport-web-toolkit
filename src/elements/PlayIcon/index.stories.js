import React from 'react';
import { storiesOf } from '@storybook/react';
import { number, boolean } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import { PlayIcon } from '../..';

const iconsStories = storiesOf('PlayIcon', module);

iconsStories.add(
  'configurable',
  withInfo()(() => <PlayIcon height={number('Height', 50)} isLoading={boolean('isLoading', false)} />)
);
