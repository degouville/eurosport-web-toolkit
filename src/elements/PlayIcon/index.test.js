import React from 'react';
import { shallow } from 'enzyme';
import PlayIcon, { StyledPlayImageSmall } from '.';

describe.only('PlayIcon', () => {
  it('should render default icon', () => {
    const wrapper = shallow(<PlayIcon className="test" height={50} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should apply icon height passed in through props', () => {
    const wrapper = shallow(<PlayIcon height={100} />);
    expect(wrapper).toHaveStyleRule('height', '100px');
  });

  it('should calculate the correct padding based off passed in height', () => {
    const wrapper = shallow(<PlayIcon height={100} />);
    expect(wrapper).toHaveStyleRule('padding', `${(100 * 0.45) / 2}px`);
  });

  it('should not output wrapper when isRounded=false', () => {
    const wrapper = shallow(<PlayIcon isRounded={false} />);
    expect(wrapper.is(StyledPlayImageSmall)).toEqual(true);
  });
});
