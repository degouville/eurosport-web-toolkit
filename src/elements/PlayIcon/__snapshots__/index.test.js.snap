// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayIcon should render default icon 1`] = `
<Styled(div)
  className="test"
  iconHeight={50}
  isLoading={false}
>
  <Styled(div)
    isLoading={false}
  />
  <img
    alt="play"
    src="play.svg"
  />
</Styled(div)>
`;
